### ISSUE TITLE
Rollout Stuck  after “Core engine process 0 ready.” if use custom plugin

### ISSUE THREAD

**Issue by wizkdc** [modelscope/ms-swift#4807]
**Describe the bug**
When I use an external plugin file, and set _--multi_turn_scheduler retool_, the program will stuck .

```
INFO 07-02 12:40:11 [backends.py:118] Directly load the compiled graph(s) for shape None from the cache, took 2.401 s
INFO 07-02 12:40:11 [monitor.py:33] torch.compile takes 4.41 s in total
INFO 07-02 12:40:12 [kv_cache_utils.py:634] GPU KV cache size: 386,000 tokens
INFO 07-02 12:40:12 [kv_cache_utils.py:637] Maximum concurrency for 20,000 tokens per request: 19.30x
INFO 07-02 12:40:33 [gpu_model_runner.py:1686] Graph capturing finished in 21 secs, took 0.50 GiB
INFO 07-02 12:40:33 [core.py:159] init engine (profile, create kv cache, warmup model) took 35.93 seconds
Unused or unrecognized kwargs: fps, return_tensors.
INFO 07-02 12:40:35 [core_client.py:439] Core engine process 0 ready.
```

Note that, I use the example plugin file and example scheduler **retool**.

If I use "math_tip_trick" or "math_tip_trick_multi_turn” which are in **swift.plugin.multi_turn**, the rollout will run successful.

What's more, if I add scheduler in **swift.plugin.multi_turn** and use it, the rollout will run successful too.

**Comment by hjh0119** [comment]
thanks for reporting

fixed in https://github.com/modelscope/ms-swift/pull/4822

**Comment by wizkdc** [comment]
thanks for your work!

**Comment by mingwu2333** [comment]
Rollout still stuck after “Core engine process 0 ready.”

`ms_swift                                 3.6.0.dev0`

`CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 
swift rollout 
    --model xxxx
    --use_async_engine true 
    --external_plugins ms-swift-main/examples/train/grpo/plugin/plugin.py 
    --multi_turn_scheduler shopping_assistant 
    --max_turns 6 
    --model_type qwen3 
    --template qwen3 
    --gpu_memory_utilization 0.85 
    --max_model_len 16384 
    --tensor_parallel_size 8 `

In plugin.py :
`multi_turns['shopping_assistant'] = ShoppingAssistantScheduler`

**Comment by MSS444** [comment]
same issue v3.7.0 dev

config:
# Training Parameters
training_params:
  # Basic Model Configuration
  rlhf_type: "grpo"  
  model: "deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B"  
  model_type: "deepseek_r1_distill"
  train_type: "lora"
  torch_dtype: "bfloat16"
  
  # Multi-Turn Specific Configuration
  # NEW-ROUND REASONING APPROACH (Default - Active)
  multi_turn_scheduler: "cpp2spec_newround"  # New-round reasoning scheduler
  max_turns: 3  # Maximum number of turns per conversation
  use_async_engine: true  # Enable async engine for better performance
  loss_scale: "newround_loss_scale"  # Loss scaling for new-round reasoning
  
  # External Plugin Configuration
  # NEW-ROUND REASONING PLUGIN (Active)
  external_plugins: "ms_swift/plugin_abc.py"
  
  # Reward Functions (Multi-Turn Aware)
  # NEW-ROUND REASONING REWARDS (Active)
  reward_funcs: 
    - "newround_format_reward"
    - "newround_bs_reward"
  reward_weights: [0.3, 0.7] 
  
  # vLLM Configuration
  use_vllm: true
  vllm_mode: "server"
  vllm_server_host: "0.0.0.0"
  vllm_server_port: 8000
  vllm_server_timeout: 300
  
  # GPU Configuration
  CUDA_VISIBLE_DEVICES_MS_SWIFT: "0"  # Adjust based on your setup
  CUDA_VISIBLE_DEVICES_VLLM: "1"      # Separate GPUs for vLLM server
  NPROC_PER_NODE: 1
  
  # vLLM Server Specific Parameters
  tensor_parallel_size: 1
  vllm_max_model_len: 8192
  gpu_memory_utilization: 0.9
  
  # Training Hyperparameters
  learning_rate: 1e-6
  warmup_ratio: 0.05
  num_train_epochs: 3
  max_steps: -1  # use epochs instead
  
  # Batch Configuration
  per_device_train_batch_size: 1  
  per_device_eval_batch_size: 1
  gradient_accumulation_steps: 1  
  gradient_checkpointing: true
  
  # LoRA Configuration
  lora_rank: 8
  lora_alpha: 32
  
  # Generation Parameters
  num_generations: 2  # Reduced for multi-turn (more expensive)
  temperature: 1.0
  top_p: 0.9
  top_k: 50
  max_length: 4096  
  max_completion_length: 2048
  
  # Evaluation and Saving
  eval_steps: 2
  save_steps: 2
  eval_strategy: "steps"
  save_total_limit: 2
  split_dataset_ratio: 0.1 
  
  # Logging Configuration
  logging_steps: 10
  log_completions: true
  report_to: "tensorboard"
  
  # Data Loading
  dataloader_num_workers: 4
  
  # Output Directories
  output_dir: "output_multiturn"
  logging_dir: "logs_multiturn"
  
  # DeepSpeed Configuration (Optional)
  deepspeed: "zero3_offload"  
  move_model_batches: 16
  ds3_gather_for_generation: false
  
  # Advanced Multi-Turn Settings
  async_generate: true
